TopMovies/API/MovieTarget.swift

```diff
@@ -31,27 +34,58 @@ class MovieTarget: StoreSubscriber {
 
 extension MovieTarget: TargetType {
   var baseURL: URL {
-    URL(string: "https://api.themoviedb.org/4")!
+    URL(string: "https://api.themoviedb.org/3/movie")!
   }
   var path: String {
-    "/list/1"
-  }
-  var method: Moya.Method {
     switch target {
-    case.marvelMovies: return .get
+    case .nowPlaying:
+      return "now_playing"
+    case .popular:
+      return "popular"
+    case .topRated:
+      return "top_rated"
+    case .upcoming:
+      return "upcoming"
     }
   }
+  var method: Moya.Method {
+    .get
+  }
   var sampleData: Data {
     Data()
   }
   var task: Task {
     switch target {
-    case.marvelMovies:
+    case let .nowPlaying(page):
+      return .requestParameters(
+        parameters: [
+          "api_key": key,
+          "language": "en",
+          "page": "\(page)"
+        ],
+        encoding: URLEncoding.default)
+    case let .popular(page):
+      return .requestParameters(
+        parameters: [
+          "api_key": key,
+          "language": "en",
+          "page": "\(page)"
+        ],
+        encoding: URLEncoding.default)
+    case let .topRated(page):
+      return .requestParameters(
+        parameters: [
+          "api_key": key,
+          "language": "en",
+          "page": "\(page)"
+        ],
+        encoding: URLEncoding.default)
```

Comment:
Looks like a duplication for me. Try to rewrite that getting rid of `switch case` entirely.